File: python/02.py
from functools import reduce

def fib():
  list = [1,2]

  while True:
    inc = list[len(list)-1] + list[len(list)-2]

    if inc > 4000000:
      break

    list.append(inc)

  return reduce(lambda acc,val: acc+val if val % 2 == 0 else acc, list)

print(fib())

File: rust/02.rs
fn fib() -> u32 {
  let mut inc;
  let mut myvec = vec![1,2];

  loop {
    inc = myvec[myvec.len()-1] + myvec[myvec.len()-2];
    println!("{}", inc);

    if inc > 4000000 {
      break;
    }

    myvec.push(inc);
  }

  return myvec.iter().fold(0, |acc, val| if val % 2 == 0 {acc + val} else {acc});
}

fn main() {
  println!("Result: {}", fib());
}
